docs/guide.md

```diff
@@ -0,0 +1,69 @@
+
+# Guide to the Dafny Core Library
+
+_THIS DOCUMENT IS A DRAFT UNDER ACTIVE DEBATE AND EDITING_
+
+A programming language library encapsulates a useful set of basic
+functionality that programmers can then incorporate in their custom
+program without having to rewrite, debug, or verify.
+
+Like other languages, the Dafny programming language and tool set includes
+libraries that provide functionality in various domains. This document describes 
+the _Dafny Core Library_. Dafny programs are copmponed of modules; to ensure
+verification soundness, Dafny imposes a strict non-circular dependency relationship
+among modules. This Core library contains functionality that may be needed
+by other libraries, but it itself is standalone.
+
+All the supported, delivered libraries will
+- be stable --- a minimum of backwards incompatible changes
+- be verified --- proved to meet specifications and tested with expected use cases,
+with each release of the library and of the Dafny toolset
+- be efficient --- efficient in both proofs and executables that use the libraries
+
+The Dafny code for the library is written to be understandable with a minimum of inline documentation
+and is open source and available [on github](https://github.com/dafny-lang/libraries). This document is
+meant to be tutorial and example-based.
+
+## Content of the Core Library
+
+The sections below describe how to use each of the principal modules of the library:
+- [Dafny.Boxes](#sec-boxes) -- containers that enable encapsulaton of success and failure results in a single datatype
+- [Bafny.BoundedInts](#sec-boundedints) -- definitions of types and cnostants for bounded integer arithmetic
+- [Dafny.FileIO}(#sec-fileio} -- basic input and output operations
+- [Dafny.Math](#sec-math) --- basic mathematical operations and lemmas
+- [Dafny.Unicode](#sec-unicode) --- tools to manipulate unicode characters and strings
+- [Dafny.Collections](#sec-collections) -- functions and lemmas about sequences, sets and maps
+- [Dafny.NonlinearArith](sec-nonlinear-arith) --  lemmas and functions for non-linear arithmetic
+- [Dafny.Relations](#sec-relations) -- predicates and lemmas about properties of functions
+
+
+## How to use the library
+
+One uses the library on the command-line by including a path to the library in the `--library` option.
```

Comment:
There will be extra steps for libraries with target language components like FileIO.

I was hoping we would have more of the features in https://github.com/dafny-lang/dafny/milestone/10 implemented to make this a smoother experience before we started labelling some of these as fully-published libraries.

Reply:
We can label it as draft-published, Dafny 4 ready code for now.

Reply:
If you do this then you need to also include the published version in your target runtime project file.